openedx/core/djangoapps/course_groups/flags.py

```diff
@@ -0,0 +1,14 @@
+"""Flags for the course_groups app."""
+from openedx.core.djangoapps.waffle_utils import CourseWaffleFlag
+
+COURSE_GROUPS_NAMESPACE = "course_groups"
+
+# .. toggle_name: course_groups.content_groups_for_teams
+# .. toggle_implementation: CourseWaffleFlag
+# .. toggle_default: False
+# .. toggle_description: This flag enables the use of content groups for teams.
+# .. toggle_use_cases: open_edx
```

Comment:
Is this off-by-default because it is new, or because some courses with teams *wouldn't* want the team-sets showing up in Studio?

To ask another way: do you think we should eventually remove the flag, or should it stay in the platform forever?

Reply:
We actually added the flag because of this comment: https://openedx.atlassian.net/wiki/spaces/COMM/pages/3885334531/Enhancements+to+Teams+to+make+learner+grouping+more+flexible?focusedCommentId=3901063179. So from the product side, this feature will be optional for the foreseeable future.
I wouldn't say _some courses with teams wouldn't want the team-sets showing up in Studio_ but _some courses with teams wouldn't want the team-sets content groups association in their Studio configuration_.

Reply:
In my opinion is little bit of both. Yes, Daniel during the product review asked for this to be optional but I also think that reducing the friction for deployment is important for the time being. Teams is not a feature that received much work in the last few years and this feature is reversing that. Better to have the capacity to turn off in case a course finds it conflicting with their usage of teams.

I like the question if this should be an option forever. My broad take is that once the feature has had real world usage we should remove the feature. Based primarily on the fact that it can always be ignored by course authors. But that is something that I would not want to block this PR on.

Reply:
> I like the question if this should be an option forever. My broad take is that once the feature has had real world usage we should remove the feature. Based primarily on the fact that it can always be ignored by course authors. But that is something that I would not want to block this PR on.

This resonates with me @felipemontoya . This waffle flag will be a configuration option that controls a configuration option 😄 

Would you mind changing the use cases to `opt_in, temporary`, and set a target removal date for "Teak"? That would be a good point to decide whether to remove the flag, or leave as an opt-in feature forever.